config/log_config_test.go

```diff
@@ -0,0 +1,123 @@
+package config
+
+import (
+	"testing"
+
+	"go.viam.com/test"
+
+	"go.viam.com/rdk/logging"
+)
+
+func TestValidatePattern(t *testing.T) {
+	test.That(t, validatePattern("robot_server.resource_manager"), test.ShouldBeTrue)
+	test.That(t, validatePattern("robot_server.resource_manager.*"), test.ShouldBeTrue)
+	test.That(t, validatePattern("robot_server.*.resource_manager"), test.ShouldBeTrue)
+	test.That(t, validatePattern("robot_server.*.*"), test.ShouldBeTrue)
+	test.That(t, validatePattern("*.resource_manager"), test.ShouldBeTrue)
+	test.That(t, validatePattern("*"), test.ShouldBeTrue)
+
+	test.That(t, validatePattern("robot_server..resource_manager"), test.ShouldBeFalse)
+	test.That(t, validatePattern("robot_server.resource_manager."), test.ShouldBeFalse)
+	test.That(t, validatePattern(".robot_server.resource_manager"), test.ShouldBeFalse)
+	test.That(t, validatePattern("robot_server.resource_manager.**"), test.ShouldBeFalse)
+	test.That(t, validatePattern("robot_server.**.resource_manager"), test.ShouldBeFalse)
+
+	test.That(t, validatePattern("_.robot_server.resource_manager"), test.ShouldBeFalse)
+	test.That(t, validatePattern("-.robot_server"), test.ShouldBeFalse)
+	test.That(t, validatePattern("robot_server.-"), test.ShouldBeFalse)
+	test.That(t, validatePattern("robot_server.-"), test.ShouldBeFalse)
+	test.That(t, validatePattern("robot_server.-.resource_manager"), test.ShouldBeFalse)
+	test.That(t, validatePattern("robot_server._.resource_manager"), test.ShouldBeFalse)
+}
+
+func TestUpdateLoggerRegistry(t *testing.T) {
+	testRegistry := map[string]logging.Logger{
+		"rdk.resource_manager":            logging.NewLogger("rdk.resource_manager"),
+		"rdk.resource_manager.modmanager": logging.NewLogger("rdk.resource_manager.modmanager"),
+		"rdk.network_traffic":             logging.NewLogger("rdk.network_traffic"),
+		"rdk.test_manager.modmanager":     logging.NewLogger("rdk.test_manager.modmanager"),
+	}
+	t.Run("basic case", func(t *testing.T) {
+		logCfg := []LoggerPatternConfig{
+			{
+				Pattern: "rdk.resource_manager",
+				Level:   "WARN",
+			},
+		}
+		newRegistry, err := UpdateLoggerRegistry(logCfg, testRegistry)
+		test.That(t, err, test.ShouldBeNil)
+
+		logger, ok := newRegistry["rdk.resource_manager"]
+		test.That(t, ok, test.ShouldBeTrue)
+		test.That(t, logger.GetLevel(), test.ShouldEqual, logging.WARN)
+
+		testRegistry["rdk.resource_manager"].SetLevel(logging.INFO)
```

Comment:
[nit] I see you're setting the level of all `testRegistry` loggers back to INFO at the end of every subtest. Can we just give each subtest their own copy of `testRegistry` (could even make a `createTestRegistry` function). If they share state as is written currently, when one fails, others may fail as the original failure will not set levels back to INFO.

Reply:
i made a helper that does the same thing, but the code is cleaner. Is it still preferred to create another copy of testRegistry?

Reply:
Gotcha. Yeah I'd still prefer a `createTestRegistry` function instead of all mutating the same `testRegistry` map.